run-tests.php

```diff
@@ -2885,6 +2886,23 @@ function generate_array_diff(array $ar1, array $ar2, bool $is_reg, array $w): ar
     $old1 = array();
     $old2 = array();
 
+    $format_diff_line = function (int $line_number, string $contents, string $diff_character) use ($colorize): string {
+        $output = sprintf("%03d%s ", $line_number, $diff_character) . $contents;
+        if ($colorize) {
+            // Light red for any line of the diff, for each line (e.g. in case post-processing removes lines).
+            // Using green for the expected lines missing from the output
+            // would cause some confusion (green could be interpreted as something the test did correctly)
+            return "\e[1;31m{$output}\e[0m";
+        }
+        return $output;
+    };
+    $format_expected_line = function (int $line_number, string $contents) use ($format_diff_line): string {
+        return $format_diff_line($line_number, $contents, '-');
+    };
+    $format_actual_line = function (int $line_number, string $contents) use ($format_diff_line): string {
+        return $format_diff_line($line_number, $contents, '+');
+    };
```

Comment:
I think I would prefer that the expected and actual line be different colours, I don't know which ones to choose however. As I personally tend to get confused by which is which >.<

Reply:
Light red and dark red? (may be unreadable on screens/terminals with poor contrast with a black background)

Colorizing just the '-' and '+' differently is one other option, but I don't know how readable that would be

Reply:
Orange and Light red might be better? But, the more I think about it, the less I know what is a good solution to this problem, so maybe just having the whole diff as one colour is fine.

Reply:
If we want to color diffs, let's color them in the usual way (red and green)...

Reply:
> This deliberately uses one color. Green would be associated with things that were done correctly.
> But both additional lines and missing lines contribute to the failure,
> so that may be confusing for something that's an error report instead of a diff
> that is meant to be applied.

That's what I initially planned to do, but I updated that before committing because it seemed misleading to use green for a line that's an error. Green definitely makes sense for git/github diffs, patch proposals, etc., but seems less clear for an error message 

For example, if context was added (`-C NUM_LINES`, like `diff -C 3`), then light green would be appropriate for the surrounding lines that actually matched the expected lines (although light grey or no colorizing would be my preference for matching context lines)

Reply:
I think you're overthinking this. The diff colors are not a value judgement. Red and green does not mean "bad" and "good", it just means "removed" and "added.

Adding some context to diffs sounds like a good idea.

Reply:
> Adding some context to diffs sounds like a good idea.

Agreed, I was wondering why that was never supported. I guess the line number of the expected(green) file could be used.

If there are substantial objections to the color, I guess an environment variable, CLI flag, or an RFC with secondary votes on the run-tests.php colorizing style could be created.

I was probably overthinking this out of concern of others objecting - I locally use `colordiff` for error reporting in another project without issues

Reply:
A screenshot of what it will look like

![run-tests-output](https://user-images.githubusercontent.com/1904430/89740824-3b98c180-da5a-11ea-8bfe-2ef59c41b9dc.png)